snuba/web/db_query.py

```diff
@@ -251,17 +264,18 @@ def _apply_thread_quota_to_clickhouse_query_settings(
     project_rate_limit_stats: Optional[RateLimitStats],
 ) -> None:
     thread_quota = query_settings.get_resource_quota()
-    if (
-        "max_threads" in clickhouse_query_settings or thread_quota is not None
-    ) and project_rate_limit_stats is not None:
+    if "max_threads" in clickhouse_query_settings or thread_quota is not None:
         maxt = (
             clickhouse_query_settings["max_threads"]
             if thread_quota is None
             else thread_quota.max_threads
         )
-        clickhouse_query_settings["max_threads"] = max(
-            1, maxt - project_rate_limit_stats.concurrent + 1
-        )
+        if project_rate_limit_stats:
+            clickhouse_query_settings["max_threads"] = max(
+                1, maxt - project_rate_limit_stats.concurrent + 1
+            )
+        else:
+            clickhouse_query_settings["max_threads"] = maxt
```

Comment:
this was a bug where if there is no project rate limiter in the query pipeline the max_threads does nothing to the query.